packages/types/src/video.ts

```diff
@@ -0,0 +1,60 @@
+// Progress callback called with percentage `0 <= p <= 1` of compression progress.
+type OnProgress = (p: number) => void
+
+export type ProcessOptions = {
+  videoName: string
+  compressedVideoName: string
+  videoCompression: number
+  chaptersConfig?: string
+  onProgress?: OnProgress
+  outputOptions?: string[]
+  videoFilters?: string
+}
+
+export type WriteVideoFrame = (data: Buffer) => void
+
+export type VideoRecording = {
+  api: RunModeVideoApi
+  controller?: BrowserVideoController
+}
```

Comment:
I'm super open to input on these APIs, especially around naming and structure. I know it's currently not the smoothest experience. They're structured the way they are to avoid a complete refactor of this part of `run.ts`, where two complex, asynchronous code paths that are interdependent require different parts of the `BrowserVideoController` at different times:

https://github.com/cypress-io/cypress/blob/4ad2279d0fbe3c9e37bc5a886a6e48819f207834/packages/server/lib/modes/run.ts#L848-L879